src/components/notebook-utils.js

```diff
@@ -73,15 +73,20 @@ export const analysisNameInput = ({ inputProps, ...props }) => h(ValidatedInput,
 
 export const tools = {
   RStudio: { label: 'RStudio', ext: 'Rmd', imageIds: ['RStudio'], defaultImageId: 'RStudio' },
-  Jupyter: { label: 'Jupyter', ext: 'ipynb', imageIds: ['terra-jupyter-bioconductor', 'terra-jupyter-bioconductor_legacy', 'terra-jupyter-hail', 'terra-jupyter-python', 'terra-jupyter-gatk', 'Pegasus', 'terra-jupyter-gatk_legacy'], defaultImageId: 'terra-jupyter-gatk' },
+  Jupyter: { label: 'Jupyter', ext: 'ipynb', imageIds: ['terra-jupyter-bioconductor', 'terra-jupyter-bioconductor_legacy', 'terra-jupyter-hail', 'terra-jupyter-python', 'terra-jupyter-gatk', 'Pegasus', 'terra-jupyter-gatk_legacy'], defaultImageId: 'terra-jupyter-gatk', isLaunchUnsupported: false },
   jupyterTerminal: { label: 'terminal' },
   spark: { label: 'spark' },
   Galaxy: { label: 'Galaxy', appType: 'GALAXY' },
-  Cromwell: { label: 'Cromwell', appType: 'CROMWELL', isAppHidden: !isCromwellAppVisible(), isPauseUnsupported: true }
+  Cromwell: { label: 'Cromwell', appType: 'CROMWELL', isAppHidden: !isCromwellAppVisible(), isPauseUnsupported: true },
+  Azure: { label: 'Azure', isAzureCompatible: true, isPauseUnsupported: true, isLaunchUnsupported: true }
 }
 
 // Returns the tools in the order that they should be displayed for Cloud Environment tools
-export const getToolsToDisplay = _.remove(tool => tool.isAppHidden)([tools.Jupyter, tools.RStudio, tools.Galaxy, tools.Cromwell])
+export const getToolsToDisplay = isAzureWorkspace => _.flow(
+  _.remove(tool => tool.isAppHidden),
+  isAzureWorkspace ? _.remove(tool => !tool.isAzureCompatible) : _.remove(tool => tool.isAzureCompatible))(
+  [tools.Jupyter, tools.RStudio, tools.Galaxy, tools.Cromwell, tools.Azure]
+)
```

Comment:
Not on you, but I don't like having to read these in the order: n, 1, 2, ..., n-1. If this is a "flow," then the river has rocks in it. I think the following indentation helps slightly (with additional suggestions):
```suggestion
export const getToolsToDisplay = isAzureWorkspace => _.flow(
  _.remove(_.get('isAppHidden')),
  _.filter(tool => tool.isAzureCompatible === isAzureWorkspace)
)([tools.Jupyter, tools.RStudio, tools.Galaxy, tools.Cromwell, tools.Azure])
```
@DataBiosphere/terra-ui-code-owners I almost wish for a utility function:
```javascript
const flowInto = (x, ...fns) => _.flow(...fns)(x)
```

Reply:
Ah I like this and the filter better. I knew there was a way to use filter over a ternary remove but my Monday brain was not grasping it :)